### PR TITLE
Feature/crm api

### LINKED ISSUE #512: [S3.5]: CRM Service
### Link to Epic

Epic: https://github.com/Open-Study-College/osc/issues/437

### Description

n/a

### Linked Tasks

#513 

_No response_

### LINKED ISSUE #513: Create mapping CRM service
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/512

### Description

Mapping data structure between third party api and OSC api

### LINKED ISSUE #514: [S3.6]: CRM endpoint
### Link to Epic

Epic: https://github.com/Open-Study-College/osc/issues/437

### Description

n/a

### Linked Tasks

#515 
#516 
#517 
#518 
#519 

_No response_

### LINKED ISSUE #516: Import CRM Library
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/514

### Description

n/a

### PR DESCRIPTION
**Pull request by Tom-Pearce**
- Closes #516 
- Closes #512 
- Closes #514 
- Closes #513 

## 📝 Description

Adds a skeleton for integrating with CRM of our choice - currently HubSpot

## ⛳️ Current behavior (updates)

None

## 🚀 New behavior

Exposes basic query to get a student record from HubSpot (contact)

### CODE REVIEW

**Review by stevenbister — APPROVED** [review]
Looks good to me! Not sure if someone with more backend knowledge would pick anything up but nothing obvious as far as I can tell

File: packages/osc-api-crm/src/types/functions.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (mattwendzina) [inline packages/osc-api-crm/src/types/functions.ts:1]: Have you purposefully added an empty file here?

**Comment by mattwendzina** [comment]
Had a look thgouh and left one comment about an empty file, but other than that LGTM 👍

**Review by hi-im-si — COMMENTED** [review]
I left a comment but my interpretation could be being applied elsewhere. please share your thoughts @Tom-Pearce

**Pull request merged**